File: 2020/day-01/v2.py
lines = open('input.txt').read().splitlines()
numbers = [int(x) for x in lines]
print(next(x * y for x in numbers for y in numbers if x + y == 2020))
print(next(x * y * z for x in numbers for y in numbers for z in numbers if x + y + z == 2020))

File: 2019/day-01/v2.hs
main = do
    text <- readFile "input.txt"
    let masses = map toInt (lines text)
    print (sum (map fuel masses))
    print (sum (map fuel' masses))
    print (sum (map fuel'' masses))
    
toInt x = read x :: Int

fuel m = m `div` 3 - 2    

fuel' m
    | f > 0 = f + fuel' f
    | otherwise = 0
    where f = fuel m

fuel'' = sum . takeWhile (>0) . tail . iterate fuel